Bento/Adapters/CollectionViewDataSource.swift

```diff
@@ -61,25 +63,23 @@ final class CollectionViewDataSource<SectionID: Hashable, ItemID: Hashable>
 
     func collectionView(_ collectionView: UICollectionView, viewForSupplementaryElementOfKind kind: String, at indexPath: IndexPath) -> UICollectionReusableView {
         let supplement = Supplement(collectionViewSupplementaryKind: kind)
+        knownSupplements.insert(supplement)
 
-        if let component = sections[indexPath.section].supplements[supplement] {
-            knownSupplements.insert(supplement)
-
-            let reuseIdentifier = component.reuseIdentifier
-            collectionView.register(CollectionViewContainerReusableView.self,
-                                    forSupplementaryViewOfKind: kind,
-                                    withReuseIdentifier: reuseIdentifier)
-            let view = collectionView.dequeueReusableSupplementaryView(
-                ofKind: kind,
-                withReuseIdentifier: reuseIdentifier,
-                for: indexPath
-            ) as! CollectionViewContainerReusableView
-
-            view.bind(component)
-            return view
-        } else {
-            return UICollectionReusableView()
-        }
+        let component = sections[indexPath.section].supplements[supplement]
+        let reuseIdentifier = component?.reuseIdentifier ?? emptyReuseIdentifier
+
+        collectionView.register(CollectionViewContainerReusableView.self,
```

Comment:
So here we are registering view every time, not sure if it has some impact on performance,

@dmcrodrigues sugested to track registered views (may be in a dictionary or something)

Reply:
AFAIK the internal logic for `register(_: forSupplementaryViewOfKind:withReuseIdentifier:)` uses a dictionary, so it wouldn't make much difference.